atom/common/api/atom_api_crash_reporter.cc

```diff
@@ -31,17 +33,39 @@ struct Converter<std::map<std::string, std::string> > {
   }
 };
 
+template<>
+struct Converter<std::vector<CrashReporter::UploadReportResult> > {
+  static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
+      const std::vector<CrashReporter::UploadReportResult>& reports) {
+    v8::Local<v8::Array> result(v8::Array::New(isolate, reports.size()));
+    for (size_t i = 0; i < reports.size(); ++i) {
+      mate::Dictionary dict(isolate, v8::Object::New(isolate));
+      dict.Set("date", reports[i].first);
+      dict.Set("id", reports[i].second);
+      v8::TryCatch try_catch;
+      result->Set(static_cast<uint32>(i), dict.GetHandle());
+      if (try_catch.HasCaught())
+        LOG(ERROR) << "Setter for index " << i << " threw an exception.";
+    }
+    return result;
+  }
+};
+
 }  // namespace mate
 
 namespace {
 
+std::vector<CrashReporter::UploadReportResult> GetUploadedReports() {
+  return (CrashReporter::GetInstance())->GetUploadedReports();
+}
+
 void Initialize(v8::Local<v8::Object> exports, v8::Local<v8::Value> unused,
                 v8::Local<v8::Context> context, void* priv) {
-  using crash_reporter::CrashReporter;
   mate::Dictionary dict(context->GetIsolate(), exports);
   dict.SetMethod("start",
                  base::Bind(&CrashReporter::Start,
                             base::Unretained(CrashReporter::GetInstance())));
+  dict.SetMethod("_getUploadedReports", &GetUploadedReports);
```

Comment:
We can use `base::Bind` to pass `CrashReporter:: GetUploadedReports` directly here instead of adding a new function.

Lines can be made shorter by using:

```c++
auto reporter = base::Unretained(CrashReporter::GetInstance()));
```